lib/protobuf/protoc/generator/message.ex

```diff
@@ -287,9 +291,21 @@ defmodule Protobuf.Protoc.Generator.Message do
     end
   end
 
+  defp merge_field_options(opts, %{options: nil}), do: opts
+
   defp merge_field_options(opts, f) do
     opts
     |> Map.put(:packed, f.options.packed)
     |> Map.put(:deprecated, f.options.deprecated)
   end
+
+  # Omit `json_name` from the options list when it matches the original field
+  # name to keep the list small. Only Proto3 has JSON support for now.
+  defp put_json_name(opts, :proto3, %{name: name, json_name: name}), do: opts
+
+  defp put_json_name(opts, :proto3, %{json_name: json_name}) do
+    Map.put(opts, :json_name, inspect(json_name))
```

Comment:
Why are we `inspect/1`ing here? What do we get as `json_name`?

Reply:
It's a string, we call `inspect` here in order to quote it correctly in the Elixir output, otherwise it will come out as a variable. There are some tests in place to avoid regressions if we eventually try to refactor this.